noxfile.py

```diff
@@ -36,11 +37,19 @@ def set_environment_variables(env_dict, session):
 
 @nox.session(name="pybamm-requires")
 def run_pybamm_requires(session):
-    """Download, compile, and install the build-time requirements for Linux and macOS: the SuiteSparse and SUNDIALS libraries."""  # noqa: E501
-    set_environment_variables(PYBAMM_ENV, session=session)
+    """Download, compile, and install the build-time requirements for Linux and macOS"""
+    force_rebuild = "--force" in session.posargs
+    sundials_path = Path("/path/to/sundials")
+    suitesparse_path = Path("/path/to/suitesparse")
+    if (sundials_path.exists() and suitesparse_path.exists()) and not force_rebuild:
+        session.log("Found existing build dependencies")
```

Comment:
```suggestion
        session.warn("Found existing build-time requirements, skipping installation. Note: run with the --force flag (nox -s pybamm-requires -- --force) to invoke re-installation.")
```

It is indeed better to run `session.log` here, but `session.warn` will display explicit warnings in yellow-coloured text which will be more visible to the user

Reply:
Also, could you similarly add a `session.log` that displays "Found pybind11" if the `pybind11/` folder exists, below?